lib/decompilerl.ex

```diff
@@ -1,11 +1,98 @@
 defmodule Decompilerl do
 
+  def summarise(module, device \\ :stdio) do
+    obtain_beam(module)
+    |> make_summary
+    |> format_summary
+    |> write_to(device)
+  end
+
   def decompile(module, device \\ :stdio) do
     obtain_beam(module)
     |> do_decompile
     |> write_to(device)
   end
 
+  defp format_summary(map) do
+    %{:file       => file,
```

Comment:
IMO it'd be more idiomatic to match (preferably in the function clause) like this:

```elixir
%{file: file, module: module, ...}
```